src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -681,6 +681,73 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
             }
         }
 
+        /// <summary>
+        /// Check that slots for member symbols indicate members of the enclosing slot's type.
+        /// </summary>
+        [Conditional("DEBUG")]
+        private void ValidateLastSlot()
+        {
+            int i = nextVariableSlot - 1;
+            if (i <= 1)
+                return;
+
+            var variable = variableBySlot[i].Symbol;
+            var containingSlot = variableBySlot[i].ContainingSlot;
+            if (containingSlot < 1)
+                return;
+
+            var containingSlotType = variableBySlot[containingSlot].Symbol.GetTypeOrReturnType().Type;
```

Comment:
>variableBySlot[containingSlot].Symbol.GetTypeOrReturnType() [](start = 37, length = 59)

We have this pattern repeated several times in the class. Consider extracting a `GetVariableType()` method.